### DISCUSSION TITLE
Port to C# [link in description]

### DISCUSSION THREAD

**Discussion by MWstudios** [artem-ogre/CDT#153] [Show and tell]
I finished the port to C#. Let me know in issues if there is a bug or a typo on my side:
https://github.com/MWstudios/C-Sharp-DT
I still have slight doubts about if some things were ported correctly. IIRC `std::unordered::map::operator[]` creates a new key if it doesn't exist, so I replicated the behavior with `Dictionary<T>()`. `detail::selection_sort()` is replaceable with `List.Sort()` (though that does a quick/heap sort) and `predicates::Epsilon<T>()` should be equal to 2^-52 (again, my assumption).

It also occasionally crashes/asserts on thin triangles when `minDistToConstraintEdge` is set to low values (below 10), though I'm unsure if it's a bug on my part.

**Comment by artem-ogre** [comment]
Hi, and thanks for the heads up @MWstudios. 
I converted it into a discussion as it is a more fitting.

**Comment by MWstudios** [comment]
Okay then. I've also ported the Ruppert refinement branch, haven't actually tested it though.

**Comment by MWstudios** [comment]
The crashing problem seems to have disappeared after I found an oversight in my code (sorting edge vertices in constructor). The source of the issue wasn't obvious until after recently updating the port to the newest version.
